src/Squidex.Infrastructure.MongoGridFs/Assets/MongoGridFsAssetStore.cs

```diff
@@ -73,13 +73,13 @@ public string GenerateSourceUrl(string id, long version, string suffix)
             {
                 var file = GetFile(name);
 
+                file.CopyTo(GetPath(id, version, suffix));
+
                 using (var stream = new MemoryStream())
                 {
-                    await bucket.DownloadToStreamByNameAsync(file.Name, stream, cancellationToken: ct);
-                    await bucket.UploadFromStreamAsync(file.Name, stream, cancellationToken: ct);
+                    await bucket.DownloadToStreamAsync(file.Name, stream, cancellationToken: ct);
```

Comment:
Can be improved with

 ```
using (var readStream = await bucket.GetDownloadStreamAsync(file))
{
	using (var writeStream = await bucket.GetUploadStreamAsync(file.Name))
	{	
		var buffer = new byte[4096]; // Should be equivalent to options.BlockSize
		var bytesRead = 0;
		
		while ((bytesRead = await readStream.ReadAsync(buffer, 0, buffer.Length)) 
		{
			await writeStream.WriteAsync(buffer, 0, bytesRead);
		}
	}
}
```

Reply:
yes, right

Reply:
but where you found GetDownloadStreamAsync?

Reply:
sry, OpenDownloadStreamAsync

Reply:
where you took options.BlockSize from ?

Reply:
I mean ChunkSizeBytes